src/libbson/doc/bson_oid_compare_unsafe.rst

```diff
@@ -0,0 +1,9 @@
+:man_page: bson_oid_compare_unsafe
+
+bson_oid_compare_unsafe()
+=========================
```

Comment:
```suggestion
=========================

Synopsis
--------

.. code-block:: c

  static inline int
  bson_oid_compare_unsafe (const bson_oid_t *oid1, const bson_oid_t *oid2);
```
Documentation pages should still include declaration of the function.

That being said, this would be the first instance of a `static` and/or `BSON_INLINE` function being included in our public documentation (although we have plenty of as-yet-unused Doxygen comment documentation), so there is no precedence here.

To that end, imo it would be more consistent for the documentation being added in this PR to be implemented as comment documentation instead. @kevinAlbs thoughts?

Reply:
> To that end, imo it would be more consistent for the documentation being added in this PR to be implemented as Doxygen comment documentation instead. @kevinAlbs thoughts?

No strong preference. If a tiebreaker opinion is needed: I have a slight preference to document in RST to include the symbols on https://mongoc.org/. Though use of Doxygen to generate documentation is planned in the near future with CDRIVER-4574.

Reply:
Then I think I am fine with reducing this comment suggestion to including the declaration of functions in the RST doc pages (with macros like `BSON_INLINE` substituted with their intended expansions, e.g. as done for `BSONCXX_EXPORT(ret-type)` in existing docs).